### PR TITLE
Fixed #28258 -- Optimized Oracle introspection by using LISTAGG.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
Before PR (on Oracle):

_migrations.test_operations.OperationTests ... **198** sec_
_inspectdb.tests ... **146** sec_
_introspection.tests ... **33** sec_
_schema.tests	... **363** sec_

after PR (on Oracle):

_migrations.test_operations.OperationTests ... **101** sec_
_inspectdb.tests ... **41** sec_
_introspection.tests ... **29** sec_
_schema.tests ... **146** sec_
```

File: django/db/backends/oracle/introspection.py [hunk 1]
```diff
@@ -171,112 +171,91 @@ def get_constraints(self, cursor, table_name):
...
-            # Record the details
-            constraints[constraint]['columns'].append(column)
+        for constraint, columns, pk, unique, check in cursor.fetchall():
+            constraints[constraint] = {
+                'columns': columns.split(','),
+                'primary_key': pk,
+                'unique': unique,
+                'foreign_key': None,
+                'check': check,
+                'index': unique,  # All uniques come with index
```
Comment (timgraham) [inline django/db/backends/oracle/introspection.py]: "with an index"

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by jtiai** [comment]
Looks good. Does this imply some restrictions about Oracle versions?

Stylistically "WITHIN GROUP" should be upper cased as in Oracle own docs.

**Comment by felixxm** [comment]
[`LISTAGG`](https://docs.oracle.com/cd/E11882_01/server.112/e41084/functions089.htm#SQLRF30030) was introduced in Oracle 11R2, so it doesn't imply any version restrictions.

**Comment by felixxm** [comment]
@jtiai Thanks for review.

**Comment by felixxm** [comment]
For comparison:
**11,418** tests -> [took **27** min](https://djangoci.com/job/django-oracle-master/database=oracle12,label=trusty,python=python3.4/952/testReport/) (before PR),
**11,424** tests -> [took **18** min](https://djangoci.com/view/PR%20Builders/job/pull-requests-oracle/lastCompletedBuild/database=oracle12,label=trusty-pr,python=python3.5/testReport/) (after PR).

**Comment by felixxm** [comment]
Merged in 8149bd00d8b8af816a683e5ecc0e204f344616f5.

**Pull request closed without merging**